src/logic/render/BaseRenderEngine.ts

```diff
@@ -1,32 +1,31 @@
-import {IRect} from "../../interfaces/IRect";
-import {store} from "../../index";
-import {ImageData} from "../../store/editor/types";
-import {ImageRepository} from "../imageRepository/ImageRepository";
+import {EditorData} from "../../data/EditorData";
+import {MouseEventUtil} from "../../utils/MouseEventUtil";
+import {MouseEventType} from "../../data/MouseEventType";
 
 export abstract class BaseRenderEngine {
     protected readonly canvas: HTMLCanvasElement;
-    protected imageRectOnCanvas: IRect;
 
-    public constructor(canvas: HTMLCanvasElement, imageRect: IRect) {
+    public constructor(canvas: HTMLCanvasElement) {
         this.canvas = canvas;
-        this.imageRectOnCanvas = imageRect;
     }
 
-    protected getActiveImageScale(): number {
-        const activeImageIndex = store.getState().editor.activeImageIndex;
-        const imageData: ImageData = store.getState().editor.imagesData[activeImageIndex];
-        const image: HTMLImageElement = ImageRepository.getById(imageData.id);
-        return image.width / this.imageRectOnCanvas.width;
+    public update(data: EditorData): void {
+        if (!!data.event) {
+            switch (MouseEventUtil.getEventType(data.event)) {
+                case MouseEventType.MOVE:
+                    return this.mouseMoveHandler(data);
```

Comment:
fajnie ten return wygląda i działa na switchu, ale kłóci mi się to z tym że te metody zwracają void'a xD
jak chcesz to zostaw. a jak przychylasz się do mojej opinii to daj "break;" zamiast "returnów"